--- a/src/backend/bpf.rs
+++ b/src/backend/bpf.rs
@@ -128,6 +128,11 @@
 // `#define AUDIT_ARCH_RISCV64  (EM_RISCV|__AUDIT_ARCH_64BIT|__AUDIT_ARCH_LE)`
 pub const AUDIT_ARCH_RISCV64: u32 = 243 | 0x8000_0000 | 0x4000_0000;
 
+// Architecture identifier for loongarch64 LE.
+// Defined as:
+// `#define AUDIT_ARCH_LOONGARCH64  (EM_LOONGARCH|__AUDIT_ARCH_64BIT|__AUDIT_ARCH_LE)`
+pub const AUDIT_ARCH_LOONGARCH64: u32 = 258 | 0x8000_0000 | 0x4000_0000;
+
 /// BPF instruction structure definition.
 // See /usr/include/linux/filter.h .
 // We cannot use the `libc::sock_filter` definition since it doesn't implement `Debug` and
--- a/src/backend/mod.rs
+++ b/src/backend/mod.rs
@@ -25,7 +25,7 @@
     SECCOMP_RET_KILL_THREAD, SECCOMP_RET_LOG, SECCOMP_RET_TRACE, SECCOMP_RET_TRAP,
 };
 
-use bpf::{ARG_NUMBER_MAX, AUDIT_ARCH_AARCH64, AUDIT_ARCH_RISCV64, AUDIT_ARCH_X86_64, BPF_MAX_LEN};
+use bpf::{ARG_NUMBER_MAX, AUDIT_ARCH_AARCH64, AUDIT_ARCH_LOONGARCH64, AUDIT_ARCH_RISCV64, AUDIT_ARCH_X86_64, BPF_MAX_LEN};
 
 pub use bpf::{sock_filter, BpfProgram, BpfProgramRef};
 
@@ -85,6 +85,8 @@
     aarch64,
     /// riscv64 arch
     riscv64,
+    /// loongarch64 arch
+    loongarch64,
 }
 
 impl TargetArch {
@@ -94,6 +96,7 @@
             TargetArch::x86_64 => AUDIT_ARCH_X86_64,
             TargetArch::aarch64 => AUDIT_ARCH_AARCH64,
             TargetArch::riscv64 => AUDIT_ARCH_RISCV64,
+            TargetArch::loongarch64 => AUDIT_ARCH_LOONGARCH64,
         }
     }
 }
@@ -105,6 +108,7 @@
             "x86_64" => Ok(TargetArch::x86_64),
             "aarch64" => Ok(TargetArch::aarch64),
             "riscv64" => Ok(TargetArch::riscv64),
+            "loongarch64" => Ok(TargetArch::loongarch64),
             _ => Err(Error::InvalidTargetArch(input.to_string())),
         }
     }
@@ -218,5 +222,14 @@
             TargetArch::try_from("RiScV64").unwrap(),
             TargetArch::riscv64
         );
+
+        assert_eq!(
+            TargetArch::try_from("loongarch64").unwrap(),
+            TargetArch::loongarch64
+        );
+        assert_eq!(
+            TargetArch::try_from("LoOnGaRcH64").unwrap(),
+            TargetArch::loongarch64
+        );
     }
 }
